central/complianceoperator/v2/report/manager/generator/report_gen_impl_test.go

```diff
@@ -105,7 +110,37 @@ func (s *ComplainceReportingTestSuite) TestProcessReportRequest() {
 		s.Require().Error(s.reportGen.ProcessReportRequest(request))
 	})
 
-	s.Run("Fail to upsert Snapshot", func() {
+	s.Run("FormatCSVReport error", func() {
+		s.snapshotDS.EXPECT().GetSnapshot(gomock.Any(), gomock.Any()).Times(1).
+			Return(&storage.ComplianceOperatorReportSnapshotV2{
+				ReportStatus: &storage.ComplianceOperatorReportStatus{},
+			}, true, nil)
+		s.resultsAggregator.EXPECT().GetReportData(gomock.Any()).Times(1).Return(&report.Results{})
+		s.formatter.EXPECT().FormatCSVReport(gomock.Any(), gomock.Any()).Times(1).Return(nil, errors.New("some error"))
+		s.snapshotDS.EXPECT().UpsertSnapshot(gomock.Any(), gomock.Any()).Times(1).
+			DoAndReturn(func(_ any, snapshot *storage.ComplianceOperatorReportSnapshotV2) error {
+				s.Require().Equal(storage.ComplianceOperatorReportStatus_FAILURE, snapshot.GetReportStatus().GetRunState())
+				return nil
+			})
+		s.Require().Error(s.reportGen.ProcessReportRequest(request))
```

Comment:
Shouldn't we assert on the type/contents of the error? The next test case makes the same assertion and I am wondering how to tell one from the other.

Reply:
(For the sake of completeness, I am copying my comment from Slack here, that I could not post before due to Github issues)

I would say, that it feels a bit more natural to have those errors wrapped and asserted on. It depends a bit on the logic, but if wrapping makes sense and adds some context to the error, then it would make it easier to investigate CI failures and potential customer diag bundles.